site/src/pages/WorkspacePage/WorkspaceScheduleControls.tsx

```diff
@@ -0,0 +1,387 @@
+  const deadlinePlusEnabled = maxDeadlineIncrease >= 1;
+  const deadlineMinusEnabled = maxDeadlineDecrease >= 1;
+
+  const onDeadlineChangeSuccess = () => {
+    displaySuccess("Updated workspace shutdown time.");
+  };
+  const onDeadlineChangeFails = (error: unknown) => {
+    displayError(
+      getErrorMessage(error, "Failed to update workspace shutdown time."),
+    );
+  };
+  const decreaseMutation = useMutation({
+    ...decreaseDeadline(workspace),
+    onSuccess: onDeadlineChangeSuccess,
+    onError: onDeadlineChangeFails,
+  });
+  const increaseMutation = useMutation({
+    ...increaseDeadline(workspace),
+    onSuccess: onDeadlineChangeSuccess,
+    onError: onDeadlineChangeFails,
+  });
+
+  return (
+    <div css={styles.scheduleValue}>
+      {isWorkspaceOn(workspace) ? (
+        <AutoStopDisplay workspace={workspace} />
+      ) : (
+        <ScheduleSettingsLink>
+          {autostartDisplay(workspace.autostart_schedule)}
+        </ScheduleSettingsLink>
+      )}
+
+      {canUpdateSchedule && canEditDeadline(workspace) && (
+        <span css={styles.scheduleControls}>
+          <Popover>
+            <PopoverTrigger>
+              <IconButton
+                disabled={!deadlineMinusEnabled}
+                size="small"
+                title="Subtract hours from deadline"
+                css={styles.scheduleButton}
+              >
+                <RemoveIcon />
+              </IconButton>
+            </PopoverTrigger>
+            <PopoverContent
+              id="schedule-sub"
+              classes={{ paper: classNames.paper }}
+              horizontal="right"
+            >
+              <DecreaseTimeContent
+                maxDeadlineDecrease={maxDeadlineDecrease}
+                onDeadlineMinus={decreaseMutation.mutate}
+              />
+            </PopoverContent>
+          </Popover>
+          <Popover>
+            <PopoverTrigger>
+              <IconButton
+                disabled={!deadlinePlusEnabled}
+                size="small"
+                title="Add hours to deadline"
+                css={styles.scheduleButton}
+              >
+                <AddIcon />
+              </IconButton>
+            </PopoverTrigger>
+            <PopoverContent
+              id="schedule-add"
+              classes={{ paper: classNames.paper }}
+              horizontal="right"
+            >
+              <AddTimeContent
+                maxDeadlineIncrease={maxDeadlineIncrease}
+                onDeadlinePlus={increaseMutation.mutate}
+              />
+            </PopoverContent>
+          </Popover>
+        </span>
+      )}
+    </div>
+  );
+};
+
+interface AddTimeContentProps {
+  maxDeadlineIncrease: number;
+  onDeadlinePlus: (value: number) => void;
+}
+
+const AddTimeContent: FC<AddTimeContentProps> = ({
+  maxDeadlineIncrease,
+  onDeadlinePlus,
+}) => {
+  const popover = usePopover();
+
+  return (
+    <>
+      <span css={styles.timePopoverTitle}>Add hours to deadline</span>
```

Comment:
I feel like, for accessibility, it might be better to mark this up as a header, and then remove the header-specific styling